capstone/src/main/java/com/google/sps/search/Trie.java

```diff
@@ -1,35 +1,121 @@
 package com.google.sps.search;
 
 import java.io.Serializable;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
+import java.util.TreeSet;
 
+/**
+ * Class representing the Trie data structure. Contains children to indicate child nodes in the data
+ * structure and whether current node is the end of the name.
+ *
+ * @author lucyqu
+ */
 public class Trie implements Serializable {
 
   private static final long serialVersionUID = 1L;
   private Map<String, Trie> children;
-  private boolean isEnd;
+  private boolean isName;
+  private Set<String> fullNames;
 
+  /**
+   * Constructor that sets the children of trie node and whether current node marks the end of the
+   * name.
+   */
   public Trie() {
-    children = new HashMap<String, Trie>();
-    isEnd = false;
+    this.children = new HashMap<String, Trie>();
+    this.isName = false;
+    this.fullNames = new HashSet<String>();
   }
 
+  /**
+   * Inserts name into trie, with full name being the last node appended to the end of the name.
+   *
+   * @param name user first name or last name
+   * @param fullName user full name.
+   */
   public void insert(String name, String fullName) {
     if (name.contentEquals("")) {
-      isEnd = true;
-      // add in the full name here
+      // indicates the end of first name or last name
+      this.isName = true;
+      fullNames.add(fullName);
```

Comment:
instead of `.isName`, couldn't you just check if `!fullNames.isEmpty()`

Reply:
I don't know if you want to expose `getFullNames()` or `getMatchingNames()` or whatever

Reply:
The reason for this comment is to me it's usually a red flag when you have a bool that represents data being present, incase the boolean and the data are ever out of sync because of a bug. Any time you're representing data with 2 different things, I always try to reduce it down to the data itself.

Reply:
But if I do a !fullNames.isEmpty() check, then it would only add the first full name. If there are several full names that I want to add to the set, I wouldn't be able to add it since fullNames would not be empty after the first insertion

Reply:
I think Andrew's idea is that when you check for this.isName on line 84 in findAll, you are just using the boolean to iterate through the fullNames set. The same can be achieved by performing the !fullNames.isEmpty() check. Therefore, you can use that to keep track of if this node contains a name instead of using the boolean. You could wrap this in a method i.e. isName() if you need to reference this outside of this file